src/wallet/rpcdump.cpp

```diff
@@ -74,19 +77,166 @@ std::string DecodeDumpString(const std::string &str) {
+
+    // Get outputs from wallet
+    std::vector<COutput> vCoins;
+    pwalletMain->AvailableCoins(vCoins, false);
+
+    // Find all outputs of the key
+    std::vector<COutput> vToSweep;
+    BOOST_FOREACH(COutput output, vCoins)
+    {
+        vector<vector<unsigned char>> vSolution;
+        txnouttype whichType;
+        if (!Solver(output.tx->vout[output.i].scriptPubKey, whichType, vSolution))
+            continue;
+
+        CKeyID keyID;
+        if (whichType == TX_PUBKEY)
+            keyID = CPubKey(vSolution[0]).GetID();
+        else
+        if (whichType == TX_PUBKEYHASH || whichType == TX_WITNESS_V0_KEYHASH)
+            keyID = CKeyID(uint160(vSolution[0]));
+
+        if (keyID == pubkey.GetID())
+            vToSweep.push_back(output);
+    }
+
+    // Add inputs to sweep transaction
+    CMutableTransaction mtx;
+    CAmount total = 0;
+    BOOST_FOREACH(COutput output, vToSweep)
+    {
+        mtx.vin.push_back(CTxIn(output.tx->GetHash(), output.i));
+        total += output.tx->vout[output.i].nValue;
+    }
+
+    if (!mtx.vin.size() || !(total > 0))
+        return false;
+
+    // Add sweep destination - fee
+    mtx.vout.push_back(CTxOut(total, GetScriptForDestination(destAddress.Get())));
+    mtx.vout[0].nValue -= 3*CWallet::GetMinimumFee(GetVirtualTransactionSize(mtx), DEFAULT_TX_CONFIRM_TARGET, mempool);
+
+    mtx.nLockTime = chainActive.Height();
+
+    // Add the private key to a keystore
+    CBasicKeyStore tempKeystore;
+    tempKeystore.AddKey(dirtyKey);
+
+    const CTransaction txConst(mtx);
+    const CKeyStore& keystoreConst = tempKeystore;
+    for (uint32_t i = 0; i < mtx.vin.size(); i++)
+    {
+        CTxIn& input = mtx.vin[i];
+        TransactionSignatureCreator creator(&keystoreConst, &txConst, i, mtx.vout[input.prevout.n].nValue);
+
+        SignatureData sigdata;
+        bool sigCreated = ProduceSignature(creator, vToSweep[i].tx->vout[input.prevout.n].scriptPubKey, sigdata);
+
+        if (sigCreated && i < mtx.vin.size()) {
+            mtx.vin[i].scriptSig = sigdata.scriptSig;
+        }
+    }
+
+    // Commit the transaction
+    const CTransaction commit(mtx);
+    CValidationState state;
+    bool fMissingInputs;
+    if (!AcceptToMemoryPool(mempool, state, commit, false, &fMissingInputs, false, maxTxFee)) {
+        if (state.IsInvalid()) {
+            throw JSONRPCError(RPC_TRANSACTION_REJECTED, strprintf("%i: %s", state.GetRejectCode(), state.GetRejectReason()));
+        } else {
+            if (fMissingInputs) {
+                throw JSONRPCError(RPC_TRANSACTION_ERROR, "Missing inputs");
+            }
+            throw JSONRPCError(RPC_TRANSACTION_ERROR, state.GetRejectReason());
+        }
+
+        LogPrintf("SweepPrivKey(): Error: Transaction not valid\n");
+        return false;
+    }
+
+    if (pwalletMain->GetBroadcastTransactions() && !g_connman)
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
```

Comment:
If I perform this check and throw the error before AcceptToMemoryPool, would that be okay?